unified-codes/libs/ui/src/lib/components/entity/molecules/EntityDetailCategoryItem.tsx

```diff
@@ -0,0 +1,24 @@
+import * as React from 'react';
+
+import { IEntity } from '@unified-codes/data';
+
+export interface EntityDetailCategoryItemProps {
+  entity: IEntity;
+}
+
+export type EntityDetailCategoryItem = React.FunctionComponent<EntityDetailCategoryItemProps>;
+
+export const EntityDetailCategoryItem: EntityDetailCategoryItem = ({ entity }) => {
+  return (
+    <li key={entity.description}>
+      {entity.description} {`(code:${entity.code})`}
+      <ul>
+        {entity.has_child?.map((child) => (
+          <EntityDetailCategoryItem entity={child} />
```

Comment:
I think this is including too much logic for a presentational component? I think this is stuff that should be controlled by the parent. E.g. unless I'm mistaken, the parent passes an entity but doesn't have control over depth of recursion?

Reply:
Also, recursion in react is a pretty dangerous path to go down because of how the render cycle works and stuff (generally would avoid components rendering themselves unless you really know what you're doing with the render cycle and stuff... which I don't!).

Reply:
Hm - I see what you're saying. Do you think it's worth specifying a maxDepth parameter on the parent? With our current dataset, it shouldn't be a problem, and the saga is currently only fetching a max of 3 children - but that could change in the future.

Reply:
Generally, I think library components which directly wrap data structures are probably to be avoided as they can be defined by a parent component (as they are often dynamic anyway...). In this instance the component exists to basically map an `entity` to HTML tags (should prob use library components for these instead, if possible). 

I think comes back to what I was saying re. composition v. OOP. OOP encourages more specific instantiations to handle specific cases, whereas composition encourages composing generic components together. Generally library components should just be presentational, i.e. think of them as functions whose purpose is just to directly display props they are given (as opposed to transformational functions which map/reduce inputs to create new data structures).

Does that make sense? I wouldn't worry about it now, I'll catch up soon with the refactor anyway!